mdpow/analysis/ensemble.py

```diff
@@ -0,0 +1,499 @@
+                                with in_dir(file, create=False):
+                                    try:
+                                        self._load_universe_from_dir(solvent)
+                                        self.add_system((solvent, dirs, file), topology=self.top[0],
+                                                        trajectory=self.trj)
+                                    except NoDataWarning:
+                                        logger.warning('Failed to load universe in %s', file)
+                                        continue
+                else:
+                    logger.error('%r directory does not exist', int_dir)
+                    raise NoDataError
+
+    def add_system(self, key, topology=None, trajectory=None, universe=None):
+        """Adds system from universe object for trajectory and topology files
+
+        Takes specified key and either existing mda.Universe object or
+        trajectory and topology path."""
+        if not universe is None:
+            self._ensemble[key] = universe
+            self._keys.append(key)
+            self._num_systems += 1
+            return
+        if isinstance(trajectory, list):
+            # setting abs paths for trajectory
+            trajectory = [os.path.abspath(trj) for trj in trajectory]
+        try:
+            self._ensemble[key] = mda.Universe(os.path.abspath(topology), trajectory)
+        except (ValueError, FileFormatWarning, NoDataError, MissingDataWarning, OSError) as err:
+            logger.warning('Error loading systems at %r', key)
+            raise NoDataWarning
+        else:
+            self._keys.append(key)
+            self._num_systems += 1
+            return
+
+    def pop(self, key):
+        """Removes and returns system at specified key.
+
+        Logs if KeyError is raised."""
+        system = self._ensemble.pop(key)
+        self._num_systems -= 1
+        return system
+
+    def select_atoms(self, *args, **kwargs):
+        """Returns :class:`~mdpow.analysis.ensemble.EnsembleAtomGroup` for
+         Universes from the :class:`~mdpow.analysis.ensemble.Ensemble`
+
+        Uses the same
+        `selection commands <https://docs.mdanalysis.org/stable/documentation_pages/selections.html>`_
+        as MDAnalysis."""
+        selections = {}
+        for key in self.keys():
+            try:
+                ag = self[key].select_atoms(*args, **kwargs)
+            except SelectionError as err:
+                logger.error("%r on system %r with selection settings %r %r", err, key, args, kwargs)
+                raise
+            else:
+                selections[key] = ag
+        return EnsembleAtomGroup(selections, ensemble_dir=self._ensemble_dir)
+
+    def select_systems(self, keys=None, solvents=None, interactions=None,
+                       lambdas=None, lambda_range=None):
+        """Select specific subset of systems and returns them in an Ensemble.
+
```

Comment:
add the parameter list here, then examples

Reply:
done